src/selectors.js

```diff
@@ -72,31 +92,102 @@ const _buildRecords = (ents, key, opts) => {
   return finalRecords;
 };
 
+const makeNestedJoins = ({ children, records, entities }) => {
+  return children.reduce((acc, child) => {
+    const parentRecords = acc.length ? acc : records;
+    const childRecords = enrichRecords(entities, child.key);
+    return joins({
+      items: parentRecords,
+      mergeItems: childRecords,
+      tableName: child.parent,
+      mergeTableName: child.key,
+    });
+  }, []);
+};
+
+const buildNestedFullRecords = ({ joinKey, children, nestedJoins }) => {
+  return nestedJoins.map((record) =>
+    buildFullRecord(
+      record,
+      joinKey,
+      children.map((c) => c.key)
+    )
+  );
+};
+
 const defaultArr = [];
 
-const buildRecords = (ents = {}, key, opts = {}) => {
-  // console.log(`All. key: ${key}`, ents, opts)
-  if (!_.isObject(opts)) opts = {};
-  const { model, defaultJoins } = getModel(key);
-  const joinKeys = utils.composeArray(opts.joins, defaultJoins);
-  let final = enrichRecords(ents, key) || defaultArr;
-
-  joinKeys.forEach((k) => {
-    const mergeItems = enrichRecords(ents, k);
-    final = joins({
-      items: final,
-      mergeItems,
+const joinsMapper = ({ joinKeys, entities, key }) => {
+  let finalRecords = enrichRecords(entities, key) || defaultArr;
+
+  joinKeys.forEach(({ key: joinKey, children }) => {
+    const records = enrichRecords(entities, joinKey);
+    const nestedJoins = makeNestedJoins({ children, records, entities });
+    const fullRecords = buildNestedFullRecords({
+      joinKey,
+      children,
+      nestedJoins,
+    });
+
+    const hasJoinsThrough = fullRecords.length;
+
+    finalRecords = joins({
```

Comment:
zouden we dit niet kunnen versimpelen door het record zelf ook in de nestedArray te verwerken? Dan zou alles via de loop in`makeNestedJoins` kunnen lopen ipv dat je hier nog los joins hoeft aan te roepen